### PR TITLE
MAINT pSMAC interface

### PR DESCRIPTION
**Pull request by mlindauer**
* FIX update of runhistory wrt to capped runs in combination with shared_model
* MAINT add option in scenario to specify a list (or a glob expression) of dictionaries from which pSMAC should read the runhistories

### CODE REVIEW

File: smac/runhistory/runhistory.py [hunk 1]
```diff
@@ -103,19 +103,27 @@ def add(self, config, cost, time,
         v = RunValue(cost, time, status, additional_info)
 
         # Each runkey is supposed to be used only once. Repeated tries to add
-        # the same runkey will be ignored silently.
-        if self.data.get(k) is None or self.data.get(k).status==StatusType.CAPPED:
-            self.data[k] = v
+        # the same runkey will be ignored silently if not capped.
+        if self.data.get(k) is None:
+            self._add(k, v, status, external_data)
+        elif status != StatusType.CAPPED and self.data[k].status == StatusType.CAPPED:
```
Comment (KEggensperger) [inline smac/runhistory/runhistory.py]: If the run is still capped but with a higher cutoff, shouldn't we store it? Can that happen?

File: smac/smbo/pSMAC.py [hunk 2]
```diff
@@ -16,8 +24,9 @@ def read(run_history, output_directory, configuration_space, logger):
         RunHistory object to be updated with run information from runhistory
         objects stored in the output directory.
 
-    output_directory : str
-        SMAC output directory. This function will search the output directory
+    output_dirs : typing.Union[str,typing.List[str]]
+        List of SMAC output directory.
```
Comment (KEggensperger) [inline smac/smbo/pSMAC.py]: directory -> directories

Comment (KEggensperger) [inline smac/smbo/pSMAC.py]: List of SMAC output directory or single SMAC output directory

File: smac/smbo/pSMAC.py [hunk 3]
```diff
@@ -29,27 +38,30 @@ def read(run_history, output_directory, configuration_space, logger):
...
-            new_numruns_in_runhistory = len(run_history.data)
-            difference = new_numruns_in_runhistory - numruns_in_runhistory
-            logger.debug('Shared model mode: Loaded %d new runs from %s' %
-                         (difference, file_in_output_directory))
-            numruns_in_runhistory = new_numruns_in_runhistory
+    if isinstance(output_dirs, str):
+        output_dirs = glob.glob(output_dirs)
+
+    for output_directory in output_dirs:
+        for file_in_output_directory in os.listdir(output_directory):
```
Comment (KEggensperger) [inline smac/smbo/pSMAC.py]: Why no glob.glob here?

Maybe first go over all entries in output_dirs to create one list with all directories found with glob

File: smac/runhistory/runhistory.py (hunk 1 above)
Comment (mlindauer) [inline smac/runhistory/runhistory.py]: thx. fixed

File: smac/smbo/pSMAC.py (hunk 3 above)
Comment (mlindauer) [inline smac/smbo/pSMAC.py]: we talked about it and I changed the docstring of the method accordingly.

**Pull request merged**